### ISSUE TITLE
Auto tag entities in the user sayings

### ISSUE THREAD

**Issue by dcalvom** [samtecspg/articulate#19]
It should be nice to recognize the entities in the text of the user sayings to help users with the task of tag entities.

**Comment by dcalvom** [comment]
In order to avoid having all the entities loaded in memory at the create intent page I was thinking on having an endpoint for highlight, something like `intent/{id}/highlight?text='I want a pizza with ham and cheese`.

This would return something like:
```
[
    {
        "value": "ham"
        "entity": "Toppings"
        "start": 20
        "end": 23
        "entityId": 30
    },
    {
        "value": "cheese"
        "entity": "Toppings"
        "start": 28
        "end": 34
        "entityId": 30
    }
]
```

So we could add those to the UI.

Of course as @wrathagom mention once it would be necessary to have a way to remove a highlight but I was just thinking about this approach.

This endpoint could also include sys entities...

**Comment by wrathagom** [comment]
Why wouldn't we use `/parse` for this?
Why is this off of the intent?

At one point in time we had discussed a manual matching of known entity values in the string, is that what `highlight` does?

I thought we were saying the solution to not loading them all into memory is to implement proper pagination.

**Comment by dcalvom** [comment]
Why wouldn't we use /parse for this?
Why is this off of the intent?

**We don't use `/parse` because at this point in time model could not be updated with newer entities. What I was thinking was on helping the user to avoid doing a manual highlight every time he is going to create a new intent**

At one point in time we had discussed a manual matching of known entity values in the string, is that what highlight does?

**Yes**

I thought we were saying the solution to not loading them all into memory is to implement proper pagination.

**By no loading in memory I'm referring to no load all agent entities in memory at the intent screen to make the comparison on the client side**

**Comment by wrathagom** [comment]
closed with #594
